src/redprl/refiner_types.fun

```diff
@@ -565,13 +554,13 @@ struct
         val Syn.SND m0 = Syn.out snd0
         val Syn.SND m1 = Syn.out snd1
 
-        val (goalTy, holeTy) = makeGoal @@ (I, H) >> CJ.SYNTH m0
-        val (goalTyB, holeTyB) = makeGoal @@ MATCH (O.MONO O.DPROD, 1, holeTy, [], [Syn.into @@ Syn.FST m0])
-        val (goalEq, _) = makeGoal @@ (I, H) >> CJ.EQ ((m0, m1), holeTy)
-        val (goalEqTy, _) = makeGoal @@ (I, H) >> CJ.EQ_TYPE (holeTyB, ty)
+        val (goalTy, holeTy) = makeSynth (I, H) m0
+        val (goalTyB, holeTyB) = makeMatch (O.MONO O.DPROD, 1, holeTy, [], [Syn.into @@ Syn.FST m0])
+        val goalEq = makeEqIfDifferent (I, H) ((m0, m1), holeTy) (* m0 well-typed *)
+        val goalEqTy = makeEqType (I, H) (holeTyB, ty) (* holeTyB type? *)
```

Comment:
Yes, I think that you can assume that `holeTyB` is a type, if I am reading these premises correctly, since `SYNTH` is meant to guarantee that its output is a type; and if `(x : A) -> B[x]` is a type, then so is `B[x]`.